README.md

```diff
@@ -1 +1,289 @@
+  let response = try await networkClient.data(request)
+  
+  // Access basic properties
+  let originalRequest = response.request
+  let payloadData: Data = response.data // might be empty. 
+  
+} catch as NetworkingError {
+
+  // Access basic properties
+  let originalRequest = error.request
+  if let response = error.response {
+    // in some cases we might have an `HTTPResponseData` value,
+    // which allows access to underlying response info
+  }
+}
+```
+
+While this is fine, it's very low level and not recommended for most use-cases. Instead applications typically which to decode payload `Data` value into a specific `Coadable` type. This is where the `Request` type can be used.
+
+`Request` is a generic value with composes the `HTTPRequestData` value, along with the ability to decode `Data` into some `Body` type. If the desired `Body` type conforms to `Decodable` this is automatic, but full customization is supported. It's even possible to decode the data to an intermediate "data-transport-object" before converting that the desired `Body` for use as an application domain type.
+
+```swift
+// Access the network client
+@Dependency(\.networkClient) var networkClient
+
+// Create a http request data value, more on this later...
+let http = HTTPRequestData(path: "hello")
+
+// Create a request, assuming that MyExpectedBody is Decodable
+let request = Request<MyExpectedBody>(http: http) // This convenience uses default JSON decoder
+
+// Await the value response
+let (value, response) = try await networkClient.value(request)
+``` 
+
+While this work okay, it's a bit fiddly having to create seemingly two request values. Instead, it is recommended to use a constrained extension on the `Request` type.
+
+```swift
+extension Request where Body == MyExpectedBody {
+  static func myBodyValue() -> Self {
+    Request(http: .init(
+      path: "hello"
+    ))
+  }
+}
+```
+
+With this in place, our code becomes,
+
+```swift
+// Access the network client
+@Dependency(\.networkClient) var networkClient
+
+// Await the value response
+let (value, response) = try await networkClient.value(.myBodyValue())
+```
+
+## NetworkClient vs APIClient
+
+The above example shows usage of a NetworkClient. None of the examples specify anything beyond a path of "hello", which is missing some key info, such as the authority. It defaults to "GET" HTTP method, and "https" scheme. All of these properties can be get/set on the request,
+
+```swift
+var request = HTTPRequestData(method: .post, authority: "my-server.com")
+request.headerFields[.accepts] = "application/json"
+request.path = "message"
+request.greeting = "Hello World"
+print(request.debugDescription) // POST https://my-server.com/message?greeting=Hello%20World
+```
+
+But of course, doing this for every request is not desirable, properties of the server should only be configured once, and this is why it is recommended to create an API Client instead of, or in addition to a Network Client.
```

Comment:
```suggestion
But of course, doing this for every request is not desirable, properties of the server should only be configured once, and this is why it is recommended to create an API Client instead of, or in addition to, a Network Client.
```